pygments/lexers/modula2.py

```diff
@@ -1559,3 +1559,13 @@ def get_tokens_unprocessed(self, text):
 
             # return result
             yield index, token, value
+
+    def analyse_text(text):
+        """Not much we can go by. (* for comments is our best guess."""
```

Comment:
that's the same for all pascal-like languages, unfortunately.

Reply:
Ok. New plan: Figure out if it's Pascal-Like first, then check if `PROCEDURE` is present (not valid Pascal), and back out if `FUNCTION` is found (not valid Modula2)